File: 10195/knights.c
#include <stdio.h>
#include <math.h>

// Accepted

int main(int argc, char *argv[])
{
	double a, b, c;
	double s, area;
	double inradius;

	while(scanf("%lf %lf %lf\n", &a, &b, &c) == 3)
	{
		s = (a + b + c) / 2;
		area = sqrt(s * (s - a) * (s - b) * (s - c));

		if(s == 0)
			inradius = 0;
		else
			inradius = area / s;

		printf("The radius of the round table is: %.3lf\n", inradius);
	}

	return 0;
}

File: 10195/knights.py
import sys

import sys
import math

for line in sys.stdin:
	x = [float(i) for i in line.strip().split()]

	s = sum(x) / 2

	a = x[0]
	b = x[1]
	c = x[2]

	a_t = math.sqrt(s * (s - a) * (s - b) * (s - c))
	inradius = a_t / s

	print("The radius of the round table is: %.3f"%(inradius))
